hack/test-extended.sh

```diff
@@ -91,16 +94,26 @@ start_server() {
 
 start_docker_registry() {
   mkdir -p ${BASETMPDIR}/.registry
-  echo "[INFO] Creating Router"
+  echo "[INFO] Creating Router ..."
   openshift ex router --create --credentials="${KUBECONFIG}" \
     --images='openshift/origin-${component}:latest' &>/dev/null
 
-  echo "[INFO] Creating Docker Registry"
+  echo "[INFO] Creating Registry ..."
   openshift ex registry --create --credentials="${KUBECONFIG}" \
     --mount-host="${BASETMPDIR}/.registry" \
     --images='openshift/origin-${component}:latest' &>/dev/null
 }
 
+pull_and_push() {
+  local image=$1
+  local registry=$2
+  echo "[INFO] Caching $image to $registry"
+  ( docker pull $image && \
```

Comment:
Actually `openshift/origin-custom-docker-builder` is part of that build and `openshift/ruby-20-centos7` should be already inside the image, so IMHO it's safer not to pull them.